test/libsolidity/SolidityNameAndTypeResolution.cpp

```diff
@@ -6224,7 +6224,7 @@ BOOST_AUTO_TEST_CASE(too_large_arrays_for_calldata)
 			}
 		}
 	)";
-	CHECK_SUCCESS_NO_WARNINGS(text);
+	CHECK_ERROR(text, TypeError, "Array is too large to be encoded as calldata.");
```

Comment:
This should not have this warning, as it is not calldata encoded for `internal`.

Reply:
Yes, it is a bug. We can investigate, but if it is too hard, it is probably not a big deal to keep it.

Reply:
I had a look at the code, and we do not have any conditions there. I think it would make sense to enforce this length restriction both for memory and for calldata. You will not be able to access the array anyway, even in memory.

Reply:
I'm pulling this into a separate PR and also changing the text to not include "calldata" so it is not confusing when using "memory".